tenjin/actions/google_search.py

```diff
@@ -0,0 +1,64 @@
+from langchain.chains import LLMChain
+from langchain.docstore.document import Document
+from langchain.prompts import PromptTemplate
+from langchain.utilities import GoogleSearchAPIWrapper
+from typing import List
+
+search = GoogleSearchAPIWrapper()
+
+template = """
+Based on the following requests, parse out the information where a Google search may be required.
+If you think that you can answer the question without a Google search, respond with NONE.
+Write the query, and respond with ONLY the query you would submit to Google. do not include quotation marks.
+
+{query}
+
+QUERY: [Write your query here]
+"""
+
+SEARCH_PROMPT = PromptTemplate(template=template, input_variables=["query"])
+
+
+class GoogleSearch:
+    """
+    Wrapper around Google Search API that returns a list of results, sources, and metadata.
+    """
+
+    def __init__(self, llm: LLMChain, k=5):
+        self.k = k
+        self.llm = llm
+
+    def run(self, query: str) -> List[dict]:
+        """run
+
+        Args:
+            query (str): English query to search for
+
+        Returns:
+            List[dict]: List of Documents including metadata
+        """
+        chain = LLMChain(llm=self.llm, prompt=SEARCH_PROMPT)
+        search_term = chain.run(query)
+
+        if search_term == "NONE":
+            return []
+
+        results = search.results(search_term, self.k)
+        sources = []
+
+        for result in results:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To create a `list`, try to use `list` comprehension instead of a loop. List comprehension is the preferred way to make a list using Python, and it&#39;s simpler and easier to understand than using a loop.

[Learn more](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)